### PR TITLE
Update webpack-dev-server to latest version

### PR DESCRIPTION
**Pull request by dacook**
Coming from ruby-land to javascript, I make the following comparison:

1. Gemfile -> Package.json
2. Gemfile.lock -> yarn.lock

Is that correct? Sometimes we specify version numbers in `Gemfile` because there are dependencies in our project (eg rails requires special work to upgrade), but most of them we let dependabot automatically update. This helps avoid the vulnerabilities that we get warned about.

We have good browser-based tests ("system specs"), so can't we treat the JS dependencies the same way? I propose we remove version numbers from `Package.json` where possible, and update packages where tests pass.
If this generates too much volume of updates, maybe we need to come up with a way to manage it more efficiently (eg test multiple updates at once). But that should be the case for Ruby updates too.

#### What? Why?

This my first try in removing a version number from Package.json, in order to:

- Close https://github.com/openfoodfoundation/openfoodnetwork/security/dependabot/46

#### What should we test?
A dev can test that the web server still works with `foreman start`. Browse the site and check the console for no new errors.

#### Release notes

Changelog Category: Technical changes / Dependency updates

### Depends on
- https://github.com/openfoodfoundation/openfoodnetwork/pull/10300

### CODE REVIEW

**Comment by dacook** [comment]
I probably should have used `[skip ci]`, because this affects only the development environment.

I checked the first failing spec and it looks like a system resource issue.
```
  Ferrum::ProcessTimeoutError:
       Browser did not produce websocket url within 20 seconds, try to increase `:process_timeout`.
```

**Comment by sigmundpetersen** [comment]
> I checked the first failing spec and it looks like a system resource issue.

Yes, this is #10018

**Comment by jibees** [comment]
> Coming from ruby-land to javascript, I make the following comparison:
> 
> Gemfile -> Package.json
> Gemfile.lock -> yarn.lock

Yes, that's it. 


> We have good browser-based tests ("system specs"), so can't we treat the JS dependencies the same way?

I used to work with JS, and it's not (as I can see) the default way of working with it. But actually, I can't tell why. Every time a package is installed, it comes with its last version, with `^`:

ie. if I run `yarn add react`, it adds in the `package.json`: 

```
"react": "^18.2.0",
```

--> this add the latest version (`18.2.0`), and will update to the latest minor (`18.3.0`) or patch version (`18.2.1`) and ignore major version (`19.x.x`)

To update package and used the `latest` version, we used to run `yarn upgrade-interactive --latest` or `yarn upgrade <package-name> --latest`: https://classic.yarnpkg.com/lang/en/docs/cli/upgrade-interactive/

So, the conclusion is: I don't know if it's a good idea or not, but I know that's not widely used in the js world.

**Review by mkllnk — APPROVED** [review]
Looking all good from my end. Dev test passes. :heavy_check_mark: 

If the JS world like to use version numbers in Package.json then we can configure Dependabot to still update (see version-stategy).

https://github.com/openfoodfoundation/openfoodnetwork/blob/895fc94a75032ca7c3616b95f7a75fdfd34fd690/.github/dependabot.yml#L11-L15

**Comment by jibees** [comment]
As commented [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/10300#issuecomment-1398145611) seems like upgrading webpacker cause us issues: https://github.com/openfoodfoundation/openfoodnetwork/pull/8709

We should be very careful if we decide to bump this version.

**Comment by mkllnk** [comment]
It would be great to resolve these issues. The current version has a critical security issue.

**Review by jibees — CHANGES_REQUESTED** [review]
I'd love to see Maikel [comments](https://github.com/openfoodfoundation/openfoodnetwork/pull/10291#pullrequestreview-1264901461) implemented ;)

I tried on [staging](https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/servers/fr-staging/deploys/925), and the PR deployed. Locally it failed, blocked by:

```
10:31:55 webpack.1 | The command moved into a separate package: @webpack-cli/serve
```

Am I the only one?

**Comment by dacook** [comment]
Hi @jibees , I saw this too, and was unsure about where it fit in before. I've created a new PR for that which is ready to review now: https://github.com/openfoodfoundation/openfoodnetwork/pull/10300

Regarding our strategy of using Package.json or not, I think we need to have more of a discussion. I think we're currently in-between options: Dependabot is configured to only change yarn.lock, but all entries in Package.json contains version numbers (as per industry standard). I've had a go at summarising this in the forum, could you please comment if this is correct and share your opinions?
https://community.openfoodnetwork.org/t/javascript-dependency-management-with-package-json/2753/3

**Comment by jibees** [comment]
It's clear to me, option1 is the best!

https://community.openfoodnetwork.org/t/javascript-dependency-management-with-package-json/2753/4?u=jibees

**Comment by dacook** [comment]
Thanks for confirming, I've created a new PR to change the Dependabot config: https://github.com/openfoodfoundation/openfoodnetwork/pull/10364

Once that is merged, Dependabot will suggest bumping `webpack-dev-server`, so I will close this PR.

**Pull request closed without merging**
